AppController/djinn.rb

```diff
@@ -1289,10 +1289,15 @@ def update_node_info_cache()
             next
           end
         end
-        new_stats << node_stats
+        Thread.current[:node_stats] = node_stats
       }
-      @cluster_stats = new_stats
     }
+    new_stats = []
+    threads.each do |t|
+      t.join
```

Comment:
@obino I am also fine with removing it I added it for time improvement. What do you think?